src/main/java/org/sonarlint/intellij/ui/tree/TreeModelBuilder.java

```diff
@@ -190,14 +189,9 @@ private AbstractNode getParent() {
   public DefaultTreeModel updateModel(Map<VirtualFile, Collection<LocalIssuePointer>> map, Predicate<VirtualFile> filePredicate) {
     this.filePredicate = filePredicate;
 
-    List<VirtualFile> toRemove = new LinkedList<>();
-    for (VirtualFile f : index.getAllFiles()) {
-      if (!map.containsKey(f)) {
-        toRemove.add(f);
-      }
-    }
-
-    toRemove.forEach(this::removeFile);
+    index.getAllFiles().stream()
+      .filter(f -> !map.containsKey(f))
+      .forEach(this::removeFile);
```

Comment:
This will cause `ConcurrentModificationException`: removing items from the underlying collection while iterating